_appmap/importer.py

```diff
@@ -193,19 +189,27 @@ def instrument_functions(filterable):
             logger.debug("  functions %s", functions)
 
             for fn_name, static_fn, fn in functions:
+                if selected_functions and fn_name not in selected_functions:
+                    continue
+
                 new_fn = cls.filter_chain.wrap(FilterableFn(filterable, fn, static_fn))
                 if fn != new_fn:
                     wrapt.wrap_function_wrapper(filterable.obj, fn_name, new_fn)
 
-        instrument_functions(FilterableMod(mod))
+        # Import Config here, to avoid circular top-level imports.
+        from .configuration import Config  # pylint: disable=import-outside-toplevel
```

Comment:
Maybe? If it was going to fail because of the way the imports are structured currently, it would already be causing tests to fail.

If the `appmap` and/or `_appmap` modules get refactored, I guess it's possible that we'll see a circular import at runtime.

I don't see a way that application code could cause a circular import here, though. I ran the listenbrainz tests with these changes, and they worked fine. So, I think I'll leave this for now. We can revisit it if it causes problems.